tfx/components/schema_gen/component.py

```diff
@@ -88,11 +93,15 @@ def __init__(
           'been renamed to "statistics" and is deprecated. Please update your '
           'usage as support for this argument will be removed soon.')
       statistics = stats
+    if exclude_splits is None:
```

Comment:
again, why exclude the eval dataset?

Note that the schema is critical in guiding the parsing of features. We've seen examples where feature "foo" is present in all the training examples, thus the generated schema says it should be parsed into a dense Tensor, however the same feature is missing in the eval examples, leading to parsing failures.

Reply:
currently, the hardcode behavior is generating a schema based on ‘train’ split. therefore, when the user doesn't provide exclude_splits, we want to keep it as its current behavior (i.e. exclude 'eval' and only use 'train'). 

for this parsing failure you mentioned, are there any solution? how can we avoid this parsing failure if a feature is only present in all training examples but not 'eval' examples?